### PR TITLE
Add new types, interfaces and enum for New Game

### PR DESCRIPTION
**Pull request by wariofdz**
### Changed:

 - Added new enum `CreateNewGameStatus.ts`. Differents status for the hook.
 - Added new type `NewGameResponse.ts`. Response from create new game.
 - Added new type `NewGameSerializedResponse.ts`. Serialized response from create new game.
 - Added new interface `GameOptions.ts`. The interface contains all options about the game.
 - Added new interface `CreateNewGameHttpRequest.ts`. The interface to send info for the API Create Game.
 - Added new interface `CreateNewGameResult.ts`. The interface with the result from the hook.
 - Added new interface `FormFieldsNewGame.ts`. The interface contains all fields from the form New Game.
 - Added new interface `FormValidationNewGameResult.ts`. The interface contains all validations fields.

### CODE REVIEW

File: packages/frontend/web-ui/src/game/models/CreateNewGameResult.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  value: string | number | GameOptions;
+}
+
+export interface CreateNewGameResult {
+  formFields: FormFieldsNewGame;
+  status: CreateNewGameStatus;
+  notifyFormFieldsFilled: () => void;
+  formValidation: FormValidationNewGameResult;
+  backendError: string | null;
+  setFormField: (params: setFormFieldsParams) => void;
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/game/models/CreateNewGameResult.ts:17]: This approach allows setting form fields with a wrong value. We need a better contract here.

File: packages/frontend/web-ui/src/game/models/CreateNewGameResult.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+export interface setFormFieldsParams {
+  name: string;
+  value: string | number | GameOptions;
+}
+
+export interface CreateNewGameResult {
+  formFields: FormFieldsNewGame;
+  status: CreateNewGameStatus;
+  notifyFormFieldsFilled: () => void;
+  formValidation: FormValidationNewGameResult;
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/game/models/CreateNewGameResult.ts:15]: Let's use the `Either` pattern here

File: packages/frontend/web-ui/src/game/models/FormValidationNewGameResult.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export interface FormValidationNewGameResult {
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/game/models/FormValidationNewGameResult.ts:1]: Let's assume this represents validation errors for the sake of the `Either` pattern usage.
```suggestion
export interface FormNewGameValidationErrorResult {
```

File: packages/frontend/web-ui/src/game/models/GameOptions.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+export interface GameOptions {
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/game/models/GameOptions.ts:1]: Let's remove this in favor of existing API models

File: packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { GameOptions } from './GameOptions';
+
+export interface CreateNewGameHttpRequest {
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts:3]: Let's rely on existing API models

**Review by notaphplover — CHANGES_REQUESTED** [review]
Let's improve typings before merging this

File: packages/frontend/web-ui/src/game/models/GameOptions.ts (hunk 4 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/game/models/GameOptions.ts:1]: Removed!

File: packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts (hunk 5 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts:3]: Changed to the API models

File: packages/frontend/web-ui/src/game/models/FormValidationNewGameResult.ts (hunk 3 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/game/models/FormValidationNewGameResult.ts:1]: Changed!

File: packages/frontend/web-ui/src/game/models/CreateNewGameResult.ts (hunk 2 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/game/models/CreateNewGameResult.ts:15]: I've changed this!

File: packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+  gameSlotsAmount: number;
+  name: string;
```
[the remark this answers is not in the document]
Comment (wariofdz) [reply packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts]: I've changed this

File: packages/frontend/web-ui/src/game/models/CreateNewGameResult.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import { CreateNewGameStatus } from './CreateNewGameStatus';
+import { FormFieldsNewGame } from './FormFieldsNewGame';
+import { FormValidationNewGameResult } from './FormValidationNewGameResult';
+import { GameOptions } from './GameOptions';
+
+export interface setFormFieldsParams {
```
[the remark this answers is not in the document]
Comment (wariofdz) [reply packages/frontend/web-ui/src/game/models/CreateNewGameResult.ts]: I've removed this interface

File: packages/frontend/web-ui/src/game/models/CreateNewGameResult.ts (hunk 1 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/game/models/CreateNewGameResult.ts]: I've removed the interface SetFormFieldsParams

File: packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import { GameOptionsV1 } from '@cornie-js/api-models/lib/models/types';
```
Comment (notaphplover) [inline packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts:1]:
```suggestion
import { models as apiModels } from '@cornie-js/api-models';
```

File: packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts (hunk 5 above)
Comment (notaphplover) [reply packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts:3]: rely on `apiModels.GameCreateQueryV1`.

**Review by notaphplover — CHANGES_REQUESTED** [review]

File: packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts (hunk 5 above)
Comment (wariofdz) [reply packages/frontend/web-ui/src/game/models/CreateNewGameHttpRequest.ts]: I've removed this interface because I didn't use anymore.

**Review by notaphplover — APPROVED** [review]

**Pull request merged**